### PR TITLE
Improved user api appstate profile img

### PR DESCRIPTION
**Pull request by jvaughn575**
So I improved the profile pic upload process. If the post of the profile image to the server fails it will not update image on the profile page. I split out the dva model into a folder in the src directory called /models/user.js. It should be now be more clear on what the user state is and how we are manipulating it in our app. 

There is still plenty of more work to be done. There are a lot of small details to make the whole process smoother. 
1. We need to persist the app's user state, so if a page reload happens we can reload/rehydrate the state. An example of how this can be done is here.
https://egghead.io/lessons/javascript-redux-persisting-the-state-to-the-local-storage

2. Restrict routes like the profile page when the user is not logged in.

3. Snackbar messages. greeting the user by name when they log in, and informing of things like if the profile image upload was a success or if it failed.

4. etc.. Its a iterative process, so I am sure there are a few details I missed in this pr. We will get them in the next one.

### CODE REVIEW

**Comment by reggieroby** [comment]
good thoughts!

i think all the followup todo's are great ideas. we should put those on jira cards and point them, see what we have time for and what should be prioritized. this will be a great time for scrum poker! :) ill drop those in a backlog. lets make a list and prioritize the teams work!

**Comment by reggieroby** [comment]
now our pull requests can have issue numbers probably! that would be cool vs jira.

File: src/components/Pages/Join.js [hunk 1]
```diff
@@ -14,16 +15,21 @@ class RegistrationForm extends React.Component {
...
     e.preventDefault();
     this.props.form.validateFieldsAndScroll((err, values) => {
       if (!err) {
         console.log('Received values of form: ', values);
         const { userName, email, password } = values;
-        addUser(userName, email, password).then(user => console.log("User Added!",user));
-        this.props.history.push('/Profile');
+        addUser(userName, email, password).then(user => {
+          if(user){
+            console.log("User Added!",user);
```
Comment (reggieroby) [inline src/components/Pages/Join.js]: i put in an issue here: https://github.com/reggieroby/travel_hive/issues/28

Comment (reggieroby) [reply src/components/Pages/Join.js]: this way you could keep all the logging in place, without it going to prod, or turning it off selectively during development debugging.

Comment (reggieroby) [reply src/components/Pages/Join.js]: yea thats cool it shows up the issue on github. lets just do scrum with github issues.

**Comment by JohnS-01** [comment]
Looks good.

File: src/App.js [hunk 2]
```diff
@@ -23,22 +23,13 @@ import {Plan} from './components/Pages/Plan';
 import {Experience} from './components/Pages/Experience';
 import {Connect} from './components/Pages/Connect';
 
+import models from './models/user.js';
```
Comment (reggieroby) [inline src/App.js]: modularizing code. great stuff!

File: src/components/Pages/Login.js [hunk 3]
```diff
@@ -12,10 +12,11 @@ class LoginForm extends React.Component {
       if (!err) {
         console.log('Received values of form: ', values);
         const { email, password } = values;
-        loginUser(email, password).then(user => {
-          console.log("Username",user);      
-          if(user){
-            this.props.dispatch({type:'isLoggedIn/yes'});   // antd dva operation to change isLoggedIn state to yes   
+        loginUser(email, password).then(user => {                
+          if(user){             
+            const base64image = String.fromCharCode.apply(null, new Uint16Array(user.profileImage.data));
```
Comment (reggieroby) [inline src/components/Pages/Login.js]: are you not storing the image as a base64 string?

**Comment by jvaughn575** [comment]
I am, but when the user logs in I send the profile image and username in
the response. When I receive the response and do res.json() the image data
is in some type of byte array format and so I convert it back to base64.

**Pull request merged**